runtime/rasdump/dmpagent.c

```diff
@@ -2757,7 +2760,38 @@ runDumpAgent(struct J9JavaVM *vm, J9RASdumpAgent * agent, J9RASdumpContext * con
 		if (gotExclusive || (agent->dumpFn != doHeapDump)) {
 			agent->prepState = *state;
 			TRIGGER_J9HOOK_VM_DUMP_START(vm->hookInterface, vm->internalVMFunctions->currentVMThread(vm), label, detail);
+#if defined(J9ZTPF)
+			struct cujvm_dmpagent_prehook_input preHookInputParms;
+			struct cujvm_dmpagent_prehook_output preHookOutputParms;
+			struct cujvm_dmpagent_posthook_input postHookInputParms;
+			struct cujvm_dmpagent_posthook_output postHookOutputParms;
+
+			memset(&preHookInputParms, '\0', sizeof(preHookInputParms));
+			memset(&preHookOutputParms, '\0', sizeof(preHookOutputParms));
+			memset(&postHookInputParms, '\0', sizeof(postHookInputParms));
+			memset(&postHookOutputParms, '\0', sizeof(postHookOutputParms));
+			preHookInputParms.agentInfo = (jvmDumpAgentInfo *)agent;
+			preHookInputParms.dumpPath = label;
+			preHookInputParms.context = (jvmDumpContext *)context;
+			preHookInputParms.vm_args = (jvmInitArgs *)vm->vmArgsArray->actualVMArgs;
+			cjvm_dumpagent_preHook(&preHookInputParms, &preHookOutputParms);
+			if (NULL != preHookOutputParms.dumpPath)  {
+				if (localLabel != label) {
+					free(label);
+				}
+				label = preHookOutputParms.dumpPath;
+			}
+#endif /* defined(J9ZTPF) */
 			retVal = runDumpFunction( agent, label, context );
+#if defined(J9ZTPF)
+			postHookInputParms.agentInfo = (jvmDumpAgentInfo *)agent;
+			postHookInputParms.dumpPath = label;
+			postHookInputParms.context = (jvmDumpContext *)context;
+			postHookInputParms.vm_args = (jvmInitArgs *)vm->vmArgsArray->actualVMArgs;
+			cjvm_dumpagent_postHook(&postHookInputParms, &postHookOutputParms);
```

Comment:
What is the purpose of `postHookOutputParms`? It's not used  here.

Does `preHookOutputParms.dumpPath` refer to allocated memory? The post-hook must not release it because it is used in the end hook below. How will it be freed?

Reply:
I should have looked more closely: `preHookOutputParms.dumpPath` must be allocated by `malloc()` given the call to `free()` below on line 2842.

Reply:
Update made for moving declarations to start of block.

Reply:
So postHookOutputParms isn't used yet... more for potential use in the future.  In case something needs to get passed back to JVM in the future.